tracee-rules/signatures/golang/anti_debugging_ptraceme.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/aquasecurity/tracee/tracee-rules/types"
+)
+
+type antiDebuggingPtraceme struct {
+	cb types.SignatureHandler
+}
+
+func (sig *antiDebuggingPtraceme) Init(cb types.SignatureHandler) error {
+	sig.cb = cb
+	return nil
+}
+
+func (sig *antiDebuggingPtraceme) GetMetadata() (types.SignatureMetadata, error) {
+	return types.SignatureMetadata{
+		Name:        "detect self debugging using PTRACE_ME",
+		Description: "Process uses anti-debugging technique to block debugger",
+		Tags:        []string{"linux", "container"},
+		Properties: map[string]interface{}{
+			"Severity": 3,
+		},
+	}, nil
+}
+
+func (sig *antiDebuggingPtraceme) GetSelectedEvents() ([]types.SignatureEventSelector, error) {
+	return []types.SignatureEventSelector{{
+		Source: "tracee",
+		Name:   "ptrace",
+	}}, nil
+}
+
+func (sig *antiDebuggingPtraceme) OnEvent(e types.Event) error {
+	const PTRACE_TRACEME int64 = 0
```

Comment:
This is a bug.
Ptrace request are received as string.
Opened PR #387 to fix this

Reply:
This will not work after rebasing on main

Reply:
I don't understand, the type of ptrace request is not int64?

Reply:
oh ok, you fixed it so it will print a string. got it

Reply:
fixed